### PR TITLE
feat: View rework part 2

### PR DESCRIPTION
**Pull request by sfc-gh-jmichalak**
- adjust SDK
    - support modifying data metric functions
    - adjust handling data metric schedule - before it was generated with incorrect format
- add `data_metric_schedule` and `data_metric_functions` to resource
- remove `or_replace` (it's used automatically when `copy_grants` is set) and `tag`, with state upgrader and migration test
- use new id handlers
- add data metric function references in SDK and use them in acceptance tests helper
- remove `USE WAREHOUSE` setup from tests
- use `sdk.PolicyKind` for handling policy kinds in policy references

## Test Plan

* [x] acceptance tests

https://docs.snowflake.com/en/sql-reference/sql/create-view#examples
https://docs.snowflake.com/en/sql-reference/functions/data_metric_function_references

## TODO (next pr)
- add `columns`
- handle data metric function's schedule status
- adjust views data source

### CODE REVIEW

File: pkg/resources/view_acceptance_test.go [hunk 1]
```diff
@@ -25,44 +25,57 @@ import (
...
 	t.Cleanup(aggregationPolicyCleanup)
 
-	rowAccessPolicy2, rowAccessPolicy2Cleanup := acc.TestClient().RowAccessPolicy.CreateRowAccessPolicyWithDataType(t, sdk.DataTypeVARCHAR)
+	rowAccessPolicy2, rowAccessPolicy2Cleanup := acc.TestClient().RowAccessPolicy.CreateRowAccessPolicyWithDataType(t, sdk.DataTypeNumber)
 	t.Cleanup(rowAccessPolicy2Cleanup)
 
 	aggregationPolicy2, aggregationPolicy2Cleanup := acc.TestClient().AggregationPolicy.CreateAggregationPolicy(t)
 	t.Cleanup(aggregationPolicy2Cleanup)
 
+	functionId := sdk.NewSchemaObjectIdentifier("SNOWFLAKE", "CORE", "AVG")
```
Comment (sfc-gh-asawicki) [inline pkg/resources/view_acceptance_test.go:45]: nit: we could extract these (like we did with snowflake preexisting roles)

File: pkg/resources/view_acceptance_test.go [hunk 2]
```diff
@@ -427,23 +540,30 @@ func TestAcc_View_complete(t *testing.T) {
...
-					assert.CheckImport(importchecks.TestCheckResourceAttrInstanceState(helpers.EncodeSnowflakeID(id), "aggregation_policy.0.entity_key.#", "1")),
-					assert.CheckImport(importchecks.TestCheckResourceAttrInstanceState(helpers.EncodeSnowflakeID(id), "aggregation_policy.0.entity_key.0", "ID")),
-					assert.CheckImport(importchecks.TestCheckResourceAttrInstanceState(helpers.EncodeSnowflakeID(id), "row_access_policy.#", "1")),
-					assert.CheckImport(importchecks.TestCheckResourceAttrInstanceState(helpers.EncodeSnowflakeID(id), "row_access_policy.0.policy_name", rowAccessPolicy.ID().FullyQualifiedName())),
-					assert.CheckImport(importchecks.TestCheckResourceAttrInstanceState(helpers.EncodeSnowflakeID(id), "row_access_policy.0.on.#", "1")),
-					assert.CheckImport(importchecks.TestCheckResourceAttrInstanceState(helpers.EncodeSnowflakeID(id), "row_access_policy.0.on.0", "ID")),
+					assert.CheckImport(importchecks.TestCheckResourceAttrInstanceState(helpers.EncodeResourceIdentifier(id), "data_metric_schedule.#", "1")),
```
Comment (sfc-gh-asawicki) [inline pkg/resources/view_acceptance_test.go:552]: nit: you could encode once, extract, and reuse

File: pkg/sdk/data_metric_function_references_gen_test.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
+package sdk
+
+import "testing"
+
+func TestDataMetricFunctionReferences_GetForEntity(t *testing.T) {
```
Comment (sfc-gh-asawicki) [inline pkg/sdk/data_metric_function_references_gen_test.go:5]: there are 5 different validation results + 1 correct validation and here are only 5 tests, so by "my math" at least one is missing

File: pkg/sdk/testint/data_metric_function_references_gen_integration_test.go [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+	client := testClient(t)
+	ctx := testContext(t)
+
+	t.Run("view domain", func(t *testing.T) {
+		functionId := sdk.NewSchemaObjectIdentifier("SNOWFLAKE", "CORE", "AVG")
+		statement := "SELECT ROLE_NAME, ROLE_OWNER FROM INFORMATION_SCHEMA.APPLICABLE_ROLES"
+		view, viewCleanup := testClientHelper().View.CreateView(t, statement)
+		t.Cleanup(viewCleanup)
+
+		err := client.Views.Alter(ctx, sdk.NewAlterViewRequest(view.ID()).WithSetDataMetricSchedule(*sdk.NewViewSetDataMetricScheduleRequest("5 MINUTE")))
```
Comment (sfc-gh-asawicki) [inline pkg/sdk/testint/data_metric_function_references_gen_integration_test.go:20]: can you please document in this test that after setting it to  minutes we get a corn response and link an issue describing it?

**Review by sfc-gh-asawicki — CHANGES_REQUESTED** [review]

File: pkg/sdk/views_validations_gen.go [hunk 5]
```diff
@@ -31,21 +29,7 @@ func (opts *CreateViewOptions) validate() error {
...
-				if !ValidObjectIdentifier(columnOption.MaskingPolicy.MaskingPolicy) {
-					errs = append(errs, errInvalidIdentifier(fmt.Sprintf("CreateViewOptions.Columns[%d]", i), "MaskingPolicy"))
-				}
-			}
-			if valueSet(columnOption.ProjectionPolicy) {
-				if !ValidObjectIdentifier(columnOption.ProjectionPolicy.ProjectionPolicy) {
-					errs = append(errs, errInvalidIdentifier(fmt.Sprintf("CreateViewOptions.Columns[%d]", i), "ProjectionPolicy"))
-				}
-			}
+			errs = append(errs, ErrInvalidObjectIdentifier)
```
Comment (sfc-gh-jcieslak) [inline pkg/sdk/views_validations_gen.go:32]: Were those validations overridden by accident or all of the removed ones were not needed? (you could revert this one at least)

File: pkg/resources/view_acceptance_test.go (hunk 1 above)
Comment (sfc-gh-jmichalak) [reply pkg/resources/view_acceptance_test.go:45]: Yes, but it's an arbitrarily chosen function with no specific meaning, and we use it only in tests, so I'll leave this for now.

File: pkg/sdk/views_validations_gen.go (hunk 5 above)
Comment (sfc-gh-jmichalak) [reply pkg/sdk/views_validations_gen.go:32]: Reverted.

File: pkg/resources/view.go [hunk 6]
```diff
@@ -92,59 +82,69 @@ var viewSchema = map[string]*schema.Schema{
...
-	// 				Default:       BooleanDefault,
-	// 				Description:   booleanStringFieldDescription("Specifies that the DMF runs when a DML operation modifies the table, such as inserting a new row or deleting a row. Conflicts with `minutes` and `using_cron`."),
-	// 				ConflictsWith: []string{"data_metric_schedule.minutes", "data_metric_schedule.using_cron"},
-	// 			},
-	// 		},
-	// 	},
-	// 	Description: "Specifies the schedule to run the data metric function periodically.",
-	// },
-	// "columns": {
+	"data_metric_functions": {
```
[the remark this answers is not in the document]
Comment (sfc-gh-jmichalak) [reply pkg/resources/view.go:85]: Done.

File: pkg/sdk/testint/data_metric_function_references_gen_integration_test.go (hunk 4 above)
Comment (sfc-gh-jmichalak) [reply pkg/sdk/testint/data_metric_function_references_gen_integration_test.go:20]: Done - linked a ticket created by views team.

File: pkg/sdk/data_metric_function_references_gen_test.go (hunk 3 above)
Comment (sfc-gh-jmichalak) [reply pkg/sdk/data_metric_function_references_gen_test.go:5]: Added.

File: pkg/resources/view_acceptance_test.go (hunk 2 above)
Comment (sfc-gh-jmichalak) [reply pkg/resources/view_acceptance_test.go:552]: Done.

File: pkg/resources/view.go [hunk 7]
```diff
@@ -92,59 +82,69 @@ var viewSchema = map[string]*schema.Schema{
...
+				},
+				"on": {
+					Type:     schema.TypeSet,
+					Required: true,
+					Elem: &schema.Schema{
+						Type: schema.TypeString,
+					},
+					Description: "The table or view columns on which to associate the data metric function. The data types of the columns must match the data types of the columns specified in the data metric function definition.",
+				},
+				// TODO (next pr)
```
[the remark this answers is not in the document]
Comment (sfc-gh-jmichalak) [reply pkg/resources/view.go:104]: Done.

File: pkg/resources/doc_helpers.go [hunk 8]
```diff
@@ -13,6 +13,14 @@ func possibleValuesListed[T ~string](values []T) string {
 	return strings.Join(valuesWrapped, " | ")
 }
 
+func possibleValuesListedInt(values []int) string {
```
[the remark this answers is not in the document]
Comment (sfc-gh-jmichalak) [reply pkg/resources/doc_helpers.go:16]: Yes, done.

**Review by sfc-gh-jcieslak — APPROVED** [review]

File: pkg/resources/view_acceptance_test.go (hunk 1 above)
Comment (sfc-gh-asawicki) [reply pkg/resources/view_acceptance_test.go:45]: oky

**Review by sfc-gh-asawicki — APPROVED** [review]

**Pull request merged**